tests/planner/ompl/test_TrajectoryConversions.cpp

```diff
@@ -0,0 +1,194 @@
+TEST_F(PlannerTest, OMPLStatesRemainUnchaged)
+{
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d midwayPose(0, -2, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  // Construct a test trajectory
+  auto traj = constructTrajectory(stateSpace, interpolator, startPose, midwayPose, goalPose);
+
+  // Get the ompl state space
+  auto si = getSpaceInformation(
+      stateSpace,
+      interpolator,
+      std::move(dmetric),
+      std::move(sampler),
+      std::move(collConstraint),
+      std::move(boundsConstraint),
+      std::move(boundsProjection),
+      0.1);
+
+  auto omplTraj = aikido::planner::ompl::toOMPLTrajectory(
+      traj, si);
+  
+  // Match every point
+  auto s0 = stateSpace->createState();
+  for(size_t idx = 0; idx < traj->getNumWaypoints(); ++idx)
+  {
+    traj->evaluate(idx, s0);
+    auto r0 = s0.getSubStateHandle<R3>(0);
+    auto stateInterpolated = r0.getValue();
+
+    auto state = omplTraj.getState(idx);
+    auto stateOMPL = getTranslationalState(stateSpace, state);
+    
+    EXPECT_EIGEN_EQUAL(stateInterpolated, stateOMPL, eigenTolerance);
+  }
+}
+
+
+// Tests that toInterpolatedTrajectory function does not alter the duration
+TEST_F(PlannerTest, InterpolatedDurationRemainsUnchaged)
+{
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d midwayPose(0, -2, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  // Construct a test trajectory
+  auto traj = constructTrajectory(stateSpace, interpolator, startPose, midwayPose, goalPose);
+
+  // Get the ompl state space
+  auto si = getSpaceInformation(
+      stateSpace,
+      interpolator,
+      std::move(dmetric),
+      std::move(sampler),
+      std::move(collConstraint),
+      std::move(boundsConstraint),
+      std::move(boundsProjection),
+      0.1);
+
+  auto omplTraj = aikido::planner::ompl::toOMPLTrajectory(
+      traj, si);
+
+  auto interpolatedTraj = aikido::planner::ompl::toInterpolatedTrajectory(
+  omplTraj, interpolator);
+
+  EXPECT_TRUE(omplTraj.getStateCount() == interpolatedTraj->getNumWaypoints());
+}
+
+
+// Tests that toInterpolatedTrajectory function does not alter the waypoints
+TEST_F(PlannerTest, InterpolatedStatesRemainUnchaged)
+{
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d midwayPose(0, -2, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  // Construct a test trajectory
+  auto traj = constructTrajectory(stateSpace, interpolator, startPose, midwayPose, goalPose);
+
+  // Get the ompl state space
+  auto si = getSpaceInformation(
+      stateSpace,
+      interpolator,
+      std::move(dmetric),
+      std::move(sampler),
+      std::move(collConstraint),
+      std::move(boundsConstraint),
+      std::move(boundsProjection),
+      0.1);
+
+  auto omplTraj = aikido::planner::ompl::toOMPLTrajectory(
+      traj, si);
+
+  auto interpolatedTraj = aikido::planner::ompl::toInterpolatedTrajectory(
+  omplTraj, interpolator);
+
+  // Match every point
```

Comment:
Verify that the two trajectories have the same number of waypoints with an `ASSERT_EQUAL`. Otherwise, the test may SEGFAULT in the loop below.